src/processes.py

```diff
@@ -105,14 +108,21 @@ def start(self):
         self.startThread(self.sc_stage, 'production-stage')
         if self.flags['runOps']:
             self.startThread(self.testOpRunner, 'heartbeat')
+            self.startThread(self.ops_generator, 'ops generator')
 
         log.warning("Successfully started node %s", self.node.get_id())
 
     def sendMessage(self, message):
+        if self.node.state == NodeState.inactive:
+            return
```

Comment:
this should raise an exception since there should be no sendMessage() calls being made in the node if the other threads are stopped.